### ISSUE TITLE
SignIn with Facebook provider in redirect flow causes unpleasant for a user message

### ISSUE THREAD

**Issue by avkonst** [firebase/firebaseui-web#527]
**Is your feature request related to a problem? Please describe.**
I have got facebook provider enabled. and signInFlow is set to 'redirect'.
This is what a user sees when attempts to sign in with facebook (find the message in red in the bottom):

![image](https://user-images.githubusercontent.com/3971413/48819439-d77d2800-edb4-11e8-8cc3-719dd18af22c.png)

This message is confusing at least for a user. The worst, it might discourage to signin.

**Describe the solution you'd like**
When firebase-ui redirects to facebook, let facebook screen to choose the optimal view as it is suggested in the red message by facebook (see screenshot above).

**Describe alternatives you've considered**
SingInFlow set to `popup` does not cause this issue, however, I prefer to have `redirect` flow, because popup windows can be blocked by a browser (as far as I have seen) and some users do not see / understand why login button does not do anything.

**Comment by avkonst** [comment]
Good to know that it is invisible for users. Thanks. In this case it is
only minor usability improvement request.

**Comment by avkonst** [comment]
To reproduce you need to be a developer of an app. You also should have narrow width window (I can not see the message when the browser is full screen, but I see it when the browser window is skinny - narrow and tall)

URL: https://nextfinal.com/login

Code snippet:
```typescript
    componentDidMount() {
        const auth = FirebaseSingleton.auth();
        const providers =[
                    firebase.auth.EmailAuthProvider.PROVIDER_ID,
                    firebase.auth.FacebookAuthProvider.PROVIDER_ID,
                    firebase.auth.GoogleAuthProvider.PROVIDER_ID
                ];
        const uiConfig = {
            signInSuccessUrl: '/',
            // TODO facebook complains on the dialog page size when it is 'redirect' page
            // TODO 'popup' alternative might not be good either as popup windows maybe blocked by browsers
            // TODO pending: https://github.com/firebase/firebaseui-web/issues/527
            signInFlow: 'redirect',
            signInOptions: providers,
            // account chooser allows to pick one of the saved users on a device
            // it is not widely known, so it is disabled in order not to distract users
            // credentialHelper: firebaseui.auth.CredentialHelper.ACCOUNT_CHOOSER_COM
            credentialHelper: firebaseui.auth.CredentialHelper.NONE,
            tosUrl: this.props.tosUrl,
            privacyPolicyUrl: this.props.privacyPolicyUrl
        };
        // Wait in case the firebase UI instance is being deleted.
        // This can happen if you unmount/remount the element quickly.
        firebaseUiDeletion.then(() => {
            // Get or Create a firebaseUI instance.
            this.ui = firebaseui.auth.AuthUI.getInstance()
                || new firebaseui.auth.AuthUI(auth);
            // Render the firebaseUi Widget.
            if (this.elementRef.current) {
                this.ui.start(this.elementRef.current, uiConfig);
            } else {
                Errors.unexpected('firebase.auth div is not rendered');
            }
        });
    }
```

**Comment by avkonst** [comment]
The facebook URL, where I end up after click signin with facebook is the
following: `
https://www.facebook.com/v2.12/dialog/oauth?response_type=code%2Cgranted_scopes&client_id=270111033693957&redirect_uri=https%3A%2F%2Flogin.nextfinal.com%2F__%2Fauth%2Fhandler&scope=email%2Cpublic_profile&state=ANwZsq9r_tsMeCuAjk20naKqwPgah5frb9_ebLm0Cuf4KUeoIJrSLAET3t4nPARPdk84D1ZZwn5lronubj147lOE0x7RBQ39BcrB5iNyHfIManX2MuyuKAaKXBzQ7kpnXRnZLoPSSlh74u8Y07GQSvDMS7TUiemQBRizXmfokPOJEZnPRws-vzThYJgTHZSPcS8P-RP9IoF2jb4GukTOmb7iknjwQ676i6cCskVnpAyRnxIONzskAvoYXdif-1-5ZTN8fZRmiwD_rT9qYpWGdw175A8SO_1OLKQx9JEYkOK9XFAIa5JHt8os-WkmEbKuoIXyyPC6x5Stzxmu8tB6ZDfhHaBSdK72l2Q0&locale=en`

**Comment by wti806** [comment]
As you can see in the URL, we didn't pass display type to facebook. Otherwise there should be a display type in the URL as query parameter. If you try with popup mode, you will see a popup display type in URL.  In this case, I don't know why facebook shows this message.

**Comment by avkonst** [comment]
I see, it is weird. Nothing we can do about it apart from maybe reporting to facebook.

**Comment by wti806** [comment]
Not sure if Facebook expected what display type if it's from a small window. But the message is quite confusing, we definitely didn't specify a explicit display type.

**Comment by dackers86** [comment]
Thank you for taking the time to submit this bug.

A new release is planned shortly, which should resolve this request. 

In the meantime, we will close this issue, if this is not resolved in the new version - please feel free to create a new issue.
